package.json

```diff
@@ -0,0 +1,18 @@
+{
+  "name": "cookbook",
+  "license": "MIT",
+  "devDependencies": {
+    "parcel": "^1.12.4",
+    "purescript": "^0.13.8",
+    "spago": "^0.15.2"
+  },
+  "config": {"recipe": "Template"},
+  "scripts": {
+    "build": "spago build",
+    "run": "spago run --main $npm_package_config_recipe.Main",
+    "serve": "parcel recipes/$npm_package_config_recipe/dev/index.html --out-dir recipes/$npm_package_config_recipe/dev-dist --open",
+    "build-dev": "NODE_ENV=development parcel build recipes/$npm_package_config_recipe/dev/index.html --out-dir recipes/$npm_package_config_recipe/dev-dist --no-minify --no-source-maps",
+    "prep-prod": "mkdir -p recipes/$npm_package_config_recipe/prod && cp recipes/$npm_package_config_recipe/dev/index.html recipes/$npm_package_config_recipe/prod/",
+    "build-prod": "npm run prep-prod --cookbook:recipe=$npm_package_config_recipe && spago bundle-app --main $npm_package_config_recipe.Main --to recipes/$npm_package_config_recipe/prod/index.js && parcel build recipes/$npm_package_config_recipe/prod/index.html --out-dir recipes/$npm_package_config_recipe/prod-dist"
+  }
```

Comment:
There's a number of questions I have here...

First, do we want to separate the difference between a production and development build? Part of me says yes because it will be a better demonstration of what to do. Part of me says no because it adds more commands one *could* run, which might be confusing for new people.

Second, if I run the `run` script for a browser-backend recipe, here's the error I get. Could we somehow improve the error a user would see to tell them something like "Hey! This recipe can't be run on the Node.js backend. Try running <correct command> instead to run it on the browser instead."
```bash
$ npm run run --cookbook:recipe=RoutingHashHalogen

> cookbook@ run /home/jordan/Programming/Git Projects/purescript-cookbook
> spago run --main $npm_package_config_recipe.Main

[info] Installation complete.

[info] Build succeeded.
/home/jordan/Programming/Git Projects/purescript-cookbook/output/Effect.Aff/foreign.js:532
                throw util.fromLeft(step);
                ^

ReferenceError: window is not defined
    at exports.window (/home/jordan/Programming/Git Projects/purescript-cookbook/output/Web.HTML/foreign.js:5:3)
    at /home/jordan/Programming/Git Projects/purescript-cookbook/output/Effect/foreign.js:12:16
    at /home/jordan/Programming/Git Projects/purescript-cookbook/output/Effect/foreign.js:12:16
    at __do (/home/jordan/Programming/Git Projects/purescript-cookbook/output/Halogen.Aff.Util/index.js:36:183)
    at runAsync (/home/jordan/Programming/Git Projects/purescript-cookbook/output/Effect.Aff/foreign.js:98:20)
    at run (/home/jordan/Programming/Git Projects/purescript-cookbook/output/Effect.Aff/foreign.js:333:22)
    at /home/jordan/Programming/Git Projects/purescript-cookbook/output/Effect.Aff/foreign.js:639:15
    at drain (/home/jordan/Programming/Git Projects/purescript-cookbook/output/Effect.Aff/foreign.js:120:9)
    at Object.enqueue (/home/jordan/Programming/Git Projects/purescript-cookbook/output/Effect.Aff/foreign.js:141:11)
    at Object.run (/home/jordan/Programming/Git Projects/purescript-cookbook/output/Effect.Aff/foreign.js:638:23)
[error] Running failed; exit code: 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! cookbook@ run: `spago run --main $npm_package_config_recipe.Main`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the cookbook@ run script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jordan/.npm/_logs/2020-06-11T01_41_20_388Z-debug.log
```

Third, I feel like each recipe should specify a `runCI` script that we use to verify that the recipe still works correctly. While doing just `spago build` gets us 80% of the way there, knowing that `parcel` made breaking changes somewhere that breaks our recipe is worth it. So, maybe each recipe could specify a `runCI.sh` file that we'll call across all recipes. If they all `return 0`, then the build passes. If not, then it fails. Thoughts?

Fourth, I feel like a command like `clean` should be included here for all recipes. Those that don't need it can just do nothing when their version is run.

Fifth, perhaps there should be a script called `initialize`, which will run `npm i && spago build`. New users can be instructed to follow these instructions:
1. `git clone` the repo
2. run `npm run initialize`
3. run `npm run <command> --cookbook:recipe=<recipe name>`

Sixth, is there a way to reconfigure the npm config, so that we can remove the `cookbook:` prefix? For example, `npm run serve --recipe=RoutingHashHalogen` rather than `npm run serve --cookbook:recipe=RoutingHashHalogen`

Seventh, I like how each recipe can use one of the following "command scripts" to run their corresponding part:
- run
- serve
- build-dev
- build-prod

I feel like we should standardize on this. I can think of a few examples
- run = run a Node-only recipe
- cli = run a Node-only recipe that takes arguments like a CLI program would
- serve-dev = run a browser-only recipe using developer-level settings
- serve-prod = run a browser-only recipe using production-level settings
- test-ci = run this script during CI to ensure the recipe is still up-to-date

Reply:
I think a lot of these points should be made into separate "enhancement" issues so we can discuss each in more detail, but here are some notes in the meantime.

> First, do we want to separate the difference between a production and development build?

Are you referring to `build-dev` vs `build-prod`? `build-dev` is only meant to be a quick way to check for `parcel serve` / `run serve` bugs (see your third point).

But I was also thinking we could just gut the production build scripts entirely, since that's not really the point of these recipes, especially since these cookbook prod scripts are a lot more complicated than what you need for a standalone project. I think it's fine for the prod stuff to just live with the framework templates.

> Second, [better errors for running browser-only example].

Yes, this would be good to include. I don't have any ideas for an elegant way to do this yet.

> Third, I feel like each recipe should specify a runCI script [to check for parcel issues]

Yep. That's what `build-dev` is for (when combined with `build-all-dev.sh`), although I didn't add that script as a CI step yet. https://github.com/milesfrain/purescript-cookbook/blob/reorg-rebase/scripts/build-all-dev.sh Was thinking of activating this in another PR.

> Fourth, ... `clean` ...

Yes, I think we can accomplish that with globs pretty easily.

> Fifth ... `initialize` ...

Maybe, but sometimes too much wrapping makes things more confusing.

> Sixth ... remove `cookbook:` prefix ...

Not that I know of. This is really annoying. I'm not very impressed with npm scripts.

> Seventh ... clearer commands ...

Yes. Thinking `npm run node` in place of `npm run run`. We could refine the others too. Maybe remove `prod`.

Reply:
I think I've opened up all issues related to this comment except for `First`.

1 = ??
2 = #10 
3 = #14
4/5/7 = #12 
6 = #13